Frends.AzureBlobStorage.UploadBlob/Frends.AzureBlobStorage.UploadBlob.Tests/UnitTestsBlock.cs

```diff
@@ -0,0 +1,637 @@
+﻿using Azure.Storage.Blobs;
+using DocumentFormat.OpenXml;
+using DocumentFormat.OpenXml.Packaging;
+using DocumentFormat.OpenXml.Spreadsheet;
+using Frends.AzureBlobStorage.UploadBlob.Definitions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Frends.AzureBlobStorage.UploadBlob.Tests;
+
+[TestClass]
+public class BlockTests
+{
+    private readonly string _connectionString = Environment.GetEnvironmentVariable("HiQ_AzureBlobStorage_ConnString");
+    private string _containerName;
+    private readonly string _appID = Environment.GetEnvironmentVariable("HiQ_AzureBlobStorage_AppID");
+    private readonly string _clientSecret = Environment.GetEnvironmentVariable("HiQ_AzureBlobStorage_ClientSecret");
+    private readonly string _tenantID = Environment.GetEnvironmentVariable("HiQ_AzureBlobStorage_TenantID");
+    private readonly string _storageAccount = "testsorage01";
+    private readonly string _testFileDir = Path.Combine(Environment.CurrentDirectory, "TestFiles");
+    private readonly string _testfile = Path.Combine(Environment.CurrentDirectory, "TestFiles", "testfile.txt");
+    private readonly string _testfile2 = Path.Combine(Environment.CurrentDirectory, "TestFiles", "testfile2.txt");
+    private readonly string _testfile3 = Path.Combine(Environment.CurrentDirectory, "TestFiles", "SearchPattern.xml");
+
```

Comment:
Please create a base test class which will set the global variables. Base class can also do the TestInitialize and TestCleanup. AppendTests can inherit the base class to get the global variables. This will make the tests more comprehensive and limits repetition.